src/macaron/repo_finder/repo_finder_deps_dev.py

```diff
@@ -71,6 +73,41 @@ def find_repo(self, purl: PackageURL) -> str:
 
         return ""
 
+    @staticmethod
+    def get_project_info(project_url: str) -> dict[str, Any] | None:
+        """Retrieve project information from deps.dev.
+
+        Parameters
+        ----------
+        project_url : str
+            The URL of the project.
+
+        Returns
+        -------
+        dict[str, Any] | None
+            The project information or None if the information could not be retrieved.
+        """
+        clean_repo_url = clean_url(project_url)
+        if clean_repo_url is None or clean_repo_url.hostname is None:
+            logger.debug("Invalid project url format: %s", project_url)
+            return None
+
+        project_key = clean_repo_url.hostname + clean_repo_url.path
+
+        request_url = f"https://api.deps.dev/v3alpha/projects/{encode(project_key, safe='')}"
+        response = send_get_http_raw(request_url)
+        if response is None or response.status_code != 200 or not response.text:
```

Comment:
With the current implementation of `send_get_http_raw` if a response object is returned (i.e. not `None`), it will have a status code of 200. So, no need to explicitly check that here. See the comment of that function for more information.